### ISSUE TITLE
Always Block when train

### ISSUE THREAD

**Issue by johnson7788** [OpenPipe/ART#401]
[train.py] _calculate_logprobs - Processing chunk 2168 to 2176
[train.py] _calculate_logprobs - Processing chunk 2176 to 2184
[train.py] _calculate_logprobs - Processing chunk 2184 to 2192
[train.py] _calculate_logprobs - Processing chunk 2192 to 2200
[train.py] _calculate_logprobs - Processing chunk 2200 to 2208
[train.py] _calculate_logprobs - Processing chunk 2208 to 2216
[train.py] _calculate_logprobs - Processing chunk 2216 to 2224
[train.py] _calculate_logprobs - Processing chunk 10152 to 10160
[train.py] _calculate_logprobs - Processing chunk 10160 to 10168
[train.py] _calculate_logprobs - Processing chunk 10168 to 10176
[train.py] _calculate_logprobs - Processing chunk 10176 to 10184
[train.py] _calculate_logprobs - Processing chunk 10184 to 10192
[train.py] _calculate_logprobs - Processing chunk 10192 to 10200
[train.py] _calculate_logprobs - Processing chunk 10200 to 10208
[train.py] _calculate_logprobs - Processing chunk 10208 to 10216
[train.py] _calculate_logprobs - Processing chunk 10216 to 10224
==((====))==  Unsloth - 2x faster free finetuning | Num GPUs used = 1
   \\   /|    Num examples = 10,000,000 | Num Epochs = 3 | Total steps = 60,000,000
O^O/ \_/ \    Batch size per device = 1 | Gradient accumulation steps = 1
\        /    Data Parallel GPUs = 1 | Total batch size (1 x 1 x 1) = 1
 "-____-"     Trainable parameters = 20,185,088 of 7,635,801,600 (0.26% trained)

DEBUG:urllib3.connectionpool:Resetting dropped connection: stats.vllm.ai
DEBUG:urllib3.connectionpool:https://stats.vllm.ai:443 "POST / HTTP/1.1" 200 None

**Comment by ninehills** [comment]
same issue:

https://github.com/OpenPipe/ART/issues/377
https://github.com/OpenPipe/ART/issues/346

**Comment by ninehills** [comment]
https://github.com/OpenPipe/ART/issues/315
